dev/TreeView/ViewModel.cpp

```diff
@@ -629,11 +632,32 @@ void ViewModel::NotifyContainerOfSelectionChange(winrt::TreeViewNode const& targ
 {
     if (m_TreeViewList)
     {
-        auto container = winrt::get_self<TreeViewList>(m_TreeViewList.get())->ContainerFromNode(targetNode);
-        if (container)
+        if (auto tvList = m_TreeViewList.get().try_as<winrt::TreeViewList>())
         {
-            winrt::TreeViewItem targetItem = container.as<winrt::TreeViewItem>();
-            winrt::get_self<TreeViewItem>(targetItem)->UpdateSelection(selectionState);
+            auto container = winrt::get_self<TreeViewList>(tvList)->ContainerFromNode(targetNode);
+            if (container)
+            {
+                winrt::TreeViewItem targetItem = container.as<winrt::TreeViewItem>();
+                winrt::get_self<TreeViewItem>(targetItem)->UpdateSelection(selectionState);
+            }
+        }
+        else if (auto nvList = m_TreeViewList.get().try_as<winrt::NavigationViewList>())
```

Comment:
I'm not so sure about those if statements. Maybe we should move the shared code to a base class and override the methods that needs different implements in sub classes? But that seems a little bit over kill for now. Let's see how this goes, if it ends up with a lot if statements, we should abstract it.

Reply:
Yeah, I wasn't sure whether this single if statement warranted sub-classing. I think this is all the NavigationView specific code I will need in this ViewModel, but if that changes, definitely something to think about.